R/html-tweak.R

```diff
@@ -324,6 +318,80 @@ badges_extract_text <- function(x) {
   xml <- xml2::read_html(x)
   badges_extract(xml)
 }
+
+activate_navbar <- function(html, path, pkg) {
+
+  navbar_haystack <- navbar_links_haystack(html, pkg)
+  path <- remove_useless_parts(path, pkg = pkg)
+
+  separate_path <- function(link) {
+    strsplit(link, "/")[[1]]
+  }
+
+  get_similarity <- function(stalk, needle) {
+    needle <- separate_path(needle)
+    stalk <- separate_path(stalk)
+
+    # Active item can't be more precise than current path/needle
+    if (length(stalk) > length(needle)) {
+      return(0)
+    }
+
+    # Active item can however be less precise than current path/needle
+    length(stalk) <- length(needle)
+    similar <- (needle == stalk)
+
+    # Any difference indicates it's not the active item
+    if (any(!similar, na.rm = TRUE)) {
+      0
+    } else {
+      sum(similar, na.rm = TRUE)
+    }
+  }
+  navbar_haystack$similar <- purrr::map_dbl(
+    navbar_haystack$links,
+    get_similarity,
+    needle = path
+  )
```

Comment:
Also I think it would be good to have a function that returns this data frame, because I think it would be good for testing.